app/controllers/api/providers_controller.rb

```diff
@@ -106,6 +127,16 @@ def destroy_provider(provider)
       action_result(false, err.to_s)
     end
 
+    def import_vm_between_providers(target_provider, source_params, target_params)
```

Comment:
If the source_params is always called that way from line 70, I'd change it to just source_vm_id, so the call on line 70 can be simplified to import_vm_between_providers(target_provider, vm_id, target_params).

Not too crazy about the method name, maybe import_vm_from_provider()

In general, was a bit confused in these changes, target_provider in the different methods, should just provider to be consistent with other actions.   Didn't know if were were targeting a different provider than this one where we're running the import_vm action on.